apps/web/src/atoms/first-load.ts

```diff
@@ -1,5 +1,21 @@
-import { atom } from 'jotai';
 import { atomWithStorage } from 'jotai/utils';
 
-export const isFirstLoadAtom = atomWithStorage<boolean>('isFirstLoad', true);
-export const openTipsAtom = atom<boolean>(false);
+export type Visibility = Record<string, boolean>;
+
+const DEFAULT_VALUE = '0.0.0';
+//atomWithStorage always uses initial value when first render
+//https://github.com/pmndrs/jotai/discussions/1737
+export const lastVersionAtom = atomWithStorage(
+  'lastVersion',
+  JSON.parse(
+    globalThis.localStorage?.getItem('lastVersion') ??
```

Comment:
Just notice that this will cause SSR hydration error. We didn’t see it just I disabled the SSR in the root element

Reply:
```suggestion
//     globalThis.localStorage?.getItem('lastVersion') ??
function getInitialValue() {
  if (typeof window !== 'undefined') {
    const storedValue = window.localStorage.getItem('lastVersion');
    if (storedValue) {
      return JSON.parse(storedValue);
    }
  }
  return DEFAULT_VALUE;
}

export const lastVersionAtom = atomWithStorage(
  'lastVersion',
  getInitialValue())
```
Can adding a window detection solve it?